src/main/java/org/springframework/data/domain/Range.java

```diff
@@ -68,13 +63,127 @@ public Range(T lowerBound, T upperBound) {
 	 * @param upperBound can be {@literal null}.
 	 * @param lowerInclusive
 	 * @param upperInclusive
+	 * @deprecated since 2.0. Use {@link Range#of(Bound, Bound)} and {@link Bound} factory methods:
+	 *             {@link Bound#inclusive(Comparable)}, {@link Bound#exclusive(Comparable)}/{@link Bound#unbounded()}.
 	 */
+	@Deprecated
 	public Range(T lowerBound, T upperBound, boolean lowerInclusive, boolean upperInclusive) {
 
+		this.lowerBound = lowerBound == null ? Bound.unbounded()
+				: lowerInclusive ? Bound.inclusive(lowerBound) : Bound.exclusive(lowerBound);
+
+		this.upperBound = upperBound == null ? Bound.unbounded()
+				: upperInclusive ? Bound.inclusive(upperBound) : Bound.exclusive(upperBound);
+	}
+
+	private Range(Bound<T> lowerBound, Bound<T> upperBound) {
```

Comment:
I guess this could be an `@RequiredArgsConstructor(access = AccessLevel.PRIVATE` if we annotated the fields with `@NonNull`?